src/prefect_server/utilities/collections.py

```diff
@@ -19,3 +20,61 @@ def chunked_iterable(iterable: Iterable, size: int):
         if not chunk:
             break
         yield chunk
+
+
+class TimedUniqueValueStore:
```

Comment:
Minor nit: Why not just `ExpiringSet` or similar?

Reply:
My naming ability is a frankly an evolving skill. I like that a lot better